plugins/modules/proxysql_cluster.py

```diff
@@ -0,0 +1,443 @@
+
+
+def load_config_to_runtime(cursor):
+    cursor.execute("LOAD PROXYSQL SERVERS TO RUNTIME")
+    return True
+
+
+class ProxySQLServer(object):
+
+    def __init__(self, module):
+        self.state = module.params["state"]
+        self.save_to_disk = module.params["save_to_disk"]
+        self.load_to_runtime = module.params["load_to_runtime"]
+
+        self.hostname = module.params["hostname"]
+        self.port = module.params["port"]
+
+        config_data_keys = [
+            "weight",
+            "comment"
+        ]
+
+        self.config_data = dict(
+            (k, module.params[k])
+            for k in config_data_keys
+        )
+
+    def check_server_config_exists(self, cursor):
+        query_string = (
+            "SELECT count(*) AS `host_count` "
+            "FROM proxysql_servers "
+            "WHERE hostname = %s "
+            "AND port = %s"
+        )
+
+        query_data = [
+            self.hostname,
+            self.port
+        ]
+
+        cursor.execute(query_string, query_data)
+        check_count = cursor.fetchone()
+        return (int(check_count['host_count']) > 0)
+
+    def check_server_config(self, cursor):
+        query_string = (
+            "SELECT count(*) AS `host_count` "
+            "FROM proxysql_servers "
+            "WHERE hostname = %s "
+            "AND port = %s"
+        )
+
+        query_data = [
+            self.hostname,
+            self.port
+        ]
+
+        for col, val in iteritems(self.config_data):
+            if val is not None:
+                query_data.append(val)
+                query_string += " AND " + col + " = %s"
+
+        cursor.execute(query_string, query_data)
+        check_count = cursor.fetchone()
+
+        if isinstance(check_count, tuple):
+            return int(check_count[0]) > 0
+
+        return (int(check_count['host_count']) > 0)
+
+    def get_server_config(self, cursor):
+        query_string = (
+            "SELECT * "
+            "FROM proxysql_servers "
+            "WHERE hostname = %s "
+            "AND port = %s"
+        )
+
+        query_data = [
+            self.hostname,
+            self.port
+        ]
+
+        cursor.execute(query_string, query_data)
+        server = cursor.fetchone()
+        return server
+
+    def create_server_config(self, cursor):
+        query_string = "INSERT INTO proxysql_servers (hostname, port"
+
+        cols = 2
+        query_data = [
+            self.hostname,
+            self.port
+        ]
+
+        for col, val in iteritems(self.config_data):
+            if val is not None:
+                cols += 1
+                query_data.append(val)
+                query_string += ", " + col
+
+        query_string += (
+            ") VALUES (" +
+            "%s ," * cols
+        )
+
+        query_string = query_string[:-2]
+        query_string += ")"
+
+        cursor.execute(query_string, query_data)
+        return True
+
+    def update_server_config(self, cursor):
```

Comment:
```suggestion
    def update_server_config(self, cursor):
```
can we pass it only once when initializing the object and use `self.cursor` everywhere?